sql/telemetry_derived/error_aggregates/query.sql

```diff
@@ -0,0 +1,185 @@
+CREATE TEMP FUNCTION
+  udf_json_extract_int_map (input STRING) AS (ARRAY(
+    SELECT
+      STRUCT(CAST(SPLIT(entry, ':')[OFFSET(0)] AS INT64) AS key,
+             CAST(SPLIT(entry, ':')[OFFSET(1)] AS INT64) AS value)
+    FROM
+      UNNEST(SPLIT(REPLACE(TRIM(input, '{}'), '"', ''), ',')) AS entry
+    WHERE
+      LENGTH(entry) > 0 ));
+CREATE TEMP FUNCTION
+  udf_json_extract_histogram (input STRING) AS (STRUCT(
+    CAST(JSON_EXTRACT_SCALAR(input, '$.bucket_count') AS INT64) AS bucket_count,
+    CAST(JSON_EXTRACT_SCALAR(input, '$.histogram_type') AS INT64) AS histogram_type,
+    CAST(JSON_EXTRACT_SCALAR(input, '$.sum') AS INT64) AS `sum`,
+    ARRAY(
+      SELECT
+        CAST(bound AS INT64)
+      FROM
+        UNNEST(SPLIT(TRIM(JSON_EXTRACT(input, '$.range'), '[]'), ',')) AS bound) AS `range`,
+    udf_json_extract_int_map(JSON_EXTRACT(input, '$.values')) AS `values` ));
+CREATE TEMP FUNCTION udf_histogram_get_sum(histogram_list ANY TYPE, target_key STRING) AS (
+  (
+    SELECT
+      udf_json_extract_histogram(value).sum
+    FROM
+      UNNEST(histogram_list)
+    WHERE
+      key = target_key
+    LIMIT
+      1
+  )
+);
+CREATE TEMP FUNCTION udf_round_timestamp_to_minute(timestamp_expression TIMESTAMP, minute INT64) AS (
+  TIMESTAMP_SECONDS(
+    CAST((FLOOR(UNIX_SECONDS(timestamp_expression) / (minute * 60)) * minute * 60) AS INT64)
+  )
+);
+--
+WITH crash_ping_agg AS (
+  SELECT
+    submission_timestamp,
+    normalized_channel AS channel,
+    environment.build.version,
+    application.display_version,
+    environment.build.build_id,
+    metadata.uri.app_name,
+    IF(metadata.uri.app_name = 'Fennec'
+       AND environment.system.os.name = 'Linux',
+       'Android',
+       environment.system.os.name) AS os,
+    environment.system.os.version AS os_version,
+    environment.build.architecture,
+    normalized_country_code AS country,
```

Comment:
We probably don't strictly some of these dimensions, though I guess it doesn't hurt to throw them in. It may help with validation.